doc/contribute_models.md

```diff
@@ -41,9 +41,65 @@ Open the browser and go to http://localhost:8888, it's a Jupyter notebook enviro
 <img src="figures/jupyter_develop.jpg">
 </p>
 
-Click into the directory `my_awesome_model` and add the `__init__.py` and your new model file, e.g. `mydnnclassifier.py`. In `__init__.py` you should expose your model classes by adding lines like `from mydnnclassifier import MyAwesomeClassifier`.
+Click into the directory `my_awesome_model` and add the `__init__.py` and your new model file, e.g. `mydnnclassifier.py`.
+
+In `mydnnclassifier.py` you should develop the model's Python code, typically a [Keras subclass model](https://www.tensorflow.org/guide/keras/custom_layers_and_models#the_model_class) like below:
+
+```python
+import tensorflow as tf
+class MyAwesomeClassifier(tf.keras.Model):
+    def __init__(self, feature_columns=None, hidden_units=[100,100]):
+        """MyAwesomeClassifier
+        :param feature_columns: feature columns.
+        :type feature_columns: list[tf.feature_column].
+        """
+        super(MyAwesomeClassifier, self).__init__()
+        self.feature_layer = None
+        if feature_columns is not None:
+            self.feature_layer = tf.keras.layers.DenseFeatures(feature_columns)
+        self.hidden_layers = []
+        for hidden_unit in hidden_units:
+            self.hidden_layers.append(tf.keras.layers.Dense(hidden_unit, activation='relu'))
+        self.prediction_layer = tf.keras.layers.Dense(1, activation='sigmoid')
+
+    def call(self, inputs, training=True):
+        if self.feature_layer is not None:
+            x = self.feature_layer(inputs)
+        else:
+            x = tf.keras.layers.Flatten()(inputs)
+        for hidden_layer in self.hidden_layers:
+            x = hidden_layer(x)
+        return self.prediction_layer(x)
+
+def optimizer(learning_rate=0.001):
+    return tf.keras.optimizers.Adagrad(lr=learning_rate)
+
+def loss(labels, output):
+    return tf.reduce_mean(tf.keras.losses.binary_crossentropy(labels, output))
+
+def prepare_prediction_column(prediction):
+    return prediction.argmax(axis=-1)
+```
+
+Note that we defined a class named `MyAwesomeClassifier` which will be used as the model definition. You can define whatever arguments in the `__init__` function of this class, these arguments can be used when you write the training SQL statement by adding `WITH argument=value, argument=value ...`. You also need to define three functions:
+
+- `optimizer`: defines the default optimizer used when training.
+- `loss`: define the default loss function used when training.
+- `prepare_prediction_column`: define how to process the prediction output.
+
+If you need to control the details of the training process or define custom models rather than a Keras model, you can define a function `sqlflow_train_loop` to implement custom model training processes:
+
+```python
+class MyAwesomeClassifier(tf.keras.Model):
+    def __init__(self, feature_columns=None):
+        ...
+    def call(self, inputs, training=True):
+        ...
+    def sqlflow_train_loop(self, dataset, epochs=1, verbose=0):
```

Comment:
Need to justify why do we need this method? I am not sure if ElasticDL can call this function.

Reply:
Model developers may need to develop models other than a Keras model, e.g. models using sklearn.

Reply:
What is the relationship between a method in a Keras class with sklearn?

Reply:
We use Keras subclass model API as SQLFlow custom model API. Yet, if we need to support custom models like SVM, kmeans or combining several models and train the model with custom training procedures like in https://github.com/sql-machine-learning/models/blob/develop/sqlflow_models/deep_embedding_cluster.py, we need a function that defines this procedure.